mcstatus/motd.py

```diff
@@ -0,0 +1,558 @@
+}
+
+
+class MinecraftColor(Enum):
+    """Enum for Color codes.
+
+    See `MineCraft wiki <https://minecraft.fandom.com/wiki/Formatting_codes#Color_codes>`_
+    for more info.
+    """
+
+    BLACK = "0"
+    DARK_BLUE = "1"
+    DARK_GREEN = "2"
+    DARK_AQUA = "3"
+    DARK_RED = "4"
+    DARK_PURPLE = "5"
+    GOLD = "6"
+    GRAY = "7"
+    DARK_GRAY = "8"
+    BLUE = "9"
+    GREEN = "a"
+    AQUA = "b"
+    RED = "c"
+    LIGHT_PURPLE = "d"
+    YELLOW = "e"
+    WHITE = "f"
+
+    # Only for bedrock
+    MINECOIN_GOLD = "g"
+
+
+MINECRAFT_COLOR_TO_HEX_BEDROCK = {
+    MinecraftColor.BLACK: ("#000000", "#000000"),
+    MinecraftColor.DARK_BLUE: ("#0000AA", "#00002A"),
+    MinecraftColor.DARK_GREEN: ("#00AA00", "#002A00"),
+    MinecraftColor.DARK_AQUA: ("#00AAAA", "#002A2A"),
+    MinecraftColor.DARK_RED: ("#AA0000", "#AA0000"),
+    MinecraftColor.DARK_PURPLE: ("#AA00AA", "#2A002A"),
+    MinecraftColor.GOLD: ("#FFAA00", "#402A00"),
+    MinecraftColor.GRAY: ("#AAAAAA", "#2A2A2A"),
+    MinecraftColor.DARK_GRAY: ("#555555", "#151515"),
+    MinecraftColor.BLUE: ("#5555FF", "#15153F"),
+    MinecraftColor.GREEN: ("#55FF55", "#153F15"),
+    MinecraftColor.AQUA: ("#55FFFF", "#153F3F"),
+    MinecraftColor.RED: ("#FF5555", "#3F1515"),
+    MinecraftColor.LIGHT_PURPLE: ("#FF55FF", "#3F153F"),
+    MinecraftColor.YELLOW: ("#FFFF55", "#3F3F15"),
+    MinecraftColor.WHITE: ("#FFFFFF", "#3F3F3F"),
+    MinecraftColor.MINECOIN_GOLD: ("#DDD605", "#373501"),
+}
+
+MINECRAFT_COLOR_TO_HEX_JAVA = MINECRAFT_COLOR_TO_HEX_BEDROCK.copy()
+MINECRAFT_COLOR_TO_HEX_JAVA[MinecraftColor.GOLD] = ("#FFAA00", "#2A2A00")
+
+MINECRAFT_COLOR_TO_ANSI = {
+    MinecraftColor.BLACK: "30",
+    MinecraftColor.DARK_BLUE: "34",
+    MinecraftColor.DARK_GREEN: "32",
+    MinecraftColor.DARK_AQUA: "36",
+    MinecraftColor.DARK_RED: "31",
+    MinecraftColor.DARK_PURPLE: "35",
+    MinecraftColor.GOLD: "33",
+    MinecraftColor.GRAY: "37",
+    MinecraftColor.DARK_GRAY: "90",
+    MinecraftColor.BLUE: "94",
+    MinecraftColor.GREEN: "92",
+    MinecraftColor.AQUA: "96",
+    MinecraftColor.RED: "91",
+    MinecraftColor.LIGHT_PURPLE: "95",
+    MinecraftColor.YELLOW: "93",
+    MinecraftColor.WHITE: "97",
+    MinecraftColor.MINECOIN_GOLD: "33",
+}
+
+
+@dataclass
+class WebColor:
+    """Raw HTML color from MOTD.
+
+    Can be found in MOTD when someone uses gradient.
+    """
+
+    hex: str
+    rgb: tuple[int, int, int]
+
+    @classmethod
+    def from_hex(cls, hex: str) -> Self:
+        """Parse hex color, from RGB.
+
+        :raises ValueError: When hex color isn't correct, so we can't generate RGB for it.
+        :returns: New ``WebColor`` instance.
+        """
+        hex = hex.lstrip("#")
+
+        try:
+            rgb = tuple(int(hex[i : i + 2], 16) for i in (0, 2, 4))
+        except ValueError:
+            raise ValueError(f"Failed to parse given hex color: {'#' + hex!r}")
+
+        return cls.from_rgb(cast("tuple[int, int, int]", rgb))
```

Comment:
The type cast here shouldn't be necessary. Pyright should infer the type of `rgb` variable as `tuple[int, ...]`, and it shouldn't mind us returning it with return type hint of `tuple[int, int, int]`

Reply:
Yes, it was PyCharm-only warning. Fixed [here](https://github.com/py-mine/mcstatus/compare/26690ce403010ab0af019cf2230d246e2920602e..ec2cacf3a1ee18ca0432a5c5e54c80ee6dbc3fcb), thanks!